lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/ChukwumaIbezimTest/BankAccountTest.java

```diff
@@ -0,0 +1,79 @@
+package com.codedifferently.lesson7.ChukwumaIbezimTest;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.codedifferently.lesson7.chukwumaibezim.*;
+import com.codedifferently.lesson7.chukwumaibezim.BankAccount.InsufficientFundsException;
+import org.junit.jupiter.api.Test;
+
+public class BankAccountTest {
+
+  @Test
+  public void testDeposit() {
+    BankAccount account =
+        new BankAccount("Jermaine Cole", 100.0, "2024-03-03", "CHECKING", 1843456889);
+    account.deposit(50.0);
+    assertEquals(150.0, account.getBalance(), 0.0);
+  }
+
+  @Test
+  public void testWithdrawSufficientFunds() throws InsufficientFundsException {
+    BankAccount account =
+        new BankAccount("Michael B. Jordan", 200.0, "2022-11-19", "SAVINGS", 987654321);
+    account.withdraw(100.0);
+    assertEquals(100.0, account.getBalance(), 0.0);
+  }
+
+  @Test
+  public void testWithdrawInsufficientFunds() {
+    BankAccount account = new BankAccount("Denzel Washington", 50.0, "2019-07-01", "CD", 135792468);
+    assertThatThrownBy(() -> account.withdraw(100.0))
+        .isInstanceOf(InsufficientFundsException.class)
+        .hasMessage("Insufficient funds to withdraw $100.0");
+  }
+
+  @Test
+  public void testTransferSufficientFunds() {
+    BankAccount sourceAccount =
+        new BankAccount("Jermaine Cole", 200.0, "2024-03-03", "CHECKING", 1843456889);
+    BankAccount targetAccount =
+        new BankAccount("Michael B. Jordan", 100.0, "2022-11-19", "SAVINGS", 987654321);
+    try {
```

Comment:
No need for try..catch here

Reply:
removed and used InsufficientFundsException